src/Dfe.PlanTech.Domain/Content/Models/Buttons/ButtonWithEntryReferenceDbEntity.cs

```diff
@@ -16,5 +16,11 @@ public class ButtonWithEntryReferenceDbEntity : ContentComponentDbEntity, IButto
     public ContentComponentDbEntity LinkToEntry { get; set; } = null!;
 
     public string LinkToEntryId { get; set; } = null!;
+
+    [DontCopyValue]
+    public string Slug { get; set; } = "";
+
+    [DontCopyValue]
+    public LinkToEntryType LinkType { get; set; } = LinkToEntryType.Unknown;
```

Comment:
Heres a wild idea that might be terrible and completely wrong - do we need linkToEntryType?
I'm thinking Slug ought to be enough, would the controller not accept a bare question link and do the right thing with it? do we need to be separating the view for where the button goes to in the page/question cases?

Reply:
We need to the subtopic for the question route; the question slug alone isn't sufficient. 

I think we _could_ do something like:

1. Only send the slug in the URL
2. Query for page by slug first - if not found by question

But this:
1. Means more DB requests unnecessarily
2. Changes the URL the user is routed to

Unless I'm being dumb and am missing something?

Reply:
ah ok - put my follow on question against the sectionSlug thing but if it doesn't work with that then good to resolve 👍

Reply:
it might, but there's potential for 2 questions (in 2 different subtopics) to have the same slug, so would cause whoopsies